phoenix-core/src/it/java/org/apache/phoenix/end2end/LogicalTableNameIT.java

```diff
@@ -0,0 +1,820 @@
+            row.add("VIEW_COL1_"+i);
+            upsertStmt.setString(6, "VIEW_COL2_"+i);
+            row.add("VIEW_COL2_"+i);
+            upsertStmt.executeUpdate();
+            result.put("PK"+i, row);
+        }
+        return result;
+    }
+
+    private void createViewAndIndex(Connection conn, String schemaName, String tableName, String viewName, String viewIndexName)
+            throws SQLException {
+        String fullTableName = SchemaUtil.getTableName(schemaName, tableName);
+        String fullViewName = SchemaUtil.getTableName(schemaName, viewName);
+        String
+                view1DDL =
+                "CREATE VIEW IF NOT EXISTS " + fullViewName + " ( VIEW_COL1 VARCHAR, VIEW_COL2 VARCHAR) AS SELECT * FROM "
+                        + fullTableName;
+        conn.createStatement().execute(view1DDL);
+        String indexDDL = "CREATE INDEX IF NOT EXISTS " + viewIndexName + " ON " + fullViewName + " (V1) include (V2, V3, VIEW_COL2) ";
+        conn.createStatement().execute(indexDDL);
+        conn.commit();
+    }
+
+    private void validateTable(Connection connection, String tableName) throws SQLException {
+        String selectTable = "SELECT PK1, V1, V2, V3 FROM " + tableName + " ORDER BY PK1 DESC";
+        ResultSet rs = connection.createStatement().executeQuery(selectTable);
+        assertTrue(rs.next());
+        assertEquals("PK3", rs.getString(1));
+        assertEquals("V13", rs.getString(2));
+        assertEquals(3, rs.getInt(3));
+        assertEquals(4, rs.getInt(4));
+        assertTrue(rs.next());
+        assertEquals("PK2", rs.getString(1));
+        assertEquals("V12", rs.getString(2));
+        assertEquals(2, rs.getInt(3));
+        assertEquals(3, rs.getInt(4));
+        assertTrue(rs.next());
+        assertEquals("PK1", rs.getString(1));
+        assertEquals("V11", rs.getString(2));
+        assertEquals(1, rs.getInt(3));
+        assertEquals(2, rs.getInt(4));
+    }
+
+    private void validateIndex(Connection connection, String tableName, boolean isViewIndex, HashMap<String, ArrayList<String>> expected) throws SQLException {
+        String selectTable = "SELECT * FROM " + tableName;
+        ResultSet rs = connection.createStatement().executeQuery(selectTable);
+        int cnt = 0;
+        while (rs.next()) {
+            String pk = rs.getString(2);
+            assertTrue(expected.containsKey(pk));
+            ArrayList<String> row = expected.get(pk);
+            assertEquals(row.get(1), rs.getString(1));
+            assertEquals(row.get(2), rs.getString(3));
+            assertEquals(row.get(3), rs.getString(4));
+            if (isViewIndex) {
+                assertEquals(row.get(5), rs.getString(5));
+            }
+            cnt++;
+        }
+        assertEquals(cnt, expected.size());
+    }
+
+    public static void renameAndDropPhysicalTable(Connection conn, String tenantId, String schema, String tableName, String physicalName) throws Exception {
```

Comment:
I don't have a case right now that requires separate methods. It is a good test to do the switch and drop so that we are sure the old table is not used. In the future, I will consider separating.